content/blog/2017/2017-01-12-scm-api-2.0-release.adoc

```diff
@@ -0,0 +1,270 @@
+* link:https://plugins.jenkins.io/branch-api[Branch API] -
+which was intended to be a plugin to hold the multi-branch functionality that was abstracted from the usage by the literate plugin.
+
+In addition, we released updates to three of the more common SCM plugins which included implementations of the SCM API:
+
+* Git plugin
+* Subversion plugin
+* Mercurial plugin
+
+While there was some interest in the literate plugin, it did not gain much traction - there are only 39 Jenkins instances running the literate plugin as of December 2016.
+
+In terms of the reusable components, we had only made a minimal implementation with some limitations:
+
+* Very basic event support - events can only trigger a re-scan of the entire repository.
+This was acceptable at the time because the only three implementations use a local cache of the remote state so re-scanning is quick.
+* No implementation of the `SCMFileSystem` API.
+As a result it is not possible for plugins like
+link:https://plugins.jenkins.io/workflow-multibranch[Pipeline Multibranch]
+to get the `Jenkinsfile` from the remote repository without needing to checkout the repository into a workspace.
+* No documentation on how plugin developers are supposed to implement the SCM API
+* No documentation on how plugin developers are supposed to consume the SCM API (if they wanted to do something like Branch API but not the same way as Branch API)
+* No documentation on how plugin developers are supposed to implement the Branch API to create their own multi-branch project types
+* No documentation on for users on how the Branch API based project types are expected to work.
+
+Roll forward to November 2015 and Jenkins Pipeline got a release of the
+link:https://plugins.jenkins.io/workflow-multibranch[Pipeline Multibranch].
+It seems that pairing Pipeline with Branch API style multi-branch is much more successful than literate - there are close to 60,000 instances running the pipeline multi-branch plugin as of December 2016.
+
+There also were two new SCM plugins implementing the SCM API:
+
+* GitHub Branch Source Plugin
+* BitBucket Branch Source Plugin
+
+Unlike the previous implementations of the SCM API, however, these plugins do not maintain a local cache of the repository state.
+Rather they make queries via the GitHub / BitBucket REST APIs on demand.
+
+The above design decision exposed one of the initial MVP compromises of the SCM API plugin: _Very basic event support_.
+Under the SCM API 1.x model, the only event that an `SCMSource` can signal is _something changed, go look at everything again_.
+When you are accessing an API that only allows 5,000 API calls per hour, performing a full scan of the entire repository just to pick up a change in one branch does not make optimum usage of that 5,000 calls/hour rate limit.
+
+So we decided that perhaps the SCM API and Branch API plugins have left their Minimum Viability Experiment state and the corresponding limitations should be addressed.
```

Comment:
This is plural and present tense so should be have. Putting "had" here completely sounds wrong to my ear